subsys/net/lib/dns/dns_sd.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020 Friedt Professional Engineering Services, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef DNS_SD_H_
+#define DNS_SD_H_
+
+#include <net/dns_sd.h>
+
+#include <stdint.h>
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Iterate over read-only DNS-SD records
+ *
+ * The ability to iterate over DNS-SD records should
+ * only ever be used by system services.
+ *
+ * Usage:
+ * @code{c}
+ * const struct _dns_sd_rec *it;
+ * DNS_SD_FOREACH(it) {
+ *   do_something_with(it);
+ * }
+ * @endcode{c}
+ */
+#define DNS_SD_FOREACH(it) \
```

Comment:
Please use `Z_STRUCT_SECTION_FOREACH()` which can be used if you define the variables using the `Z_STRUCT_SECTION_ITERABLE()` macro